client/app/HttpClient.js

```diff
@@ -0,0 +1,134 @@
+     * The headers parameter is not modified.
+     * 
+     * @param headers : the headers to combine with the default Header.
+     * @returns A new object with the full set of headers.
+     */
+    static async _defaultHeaders(headers = {}) {
+        //TODO: add Barer token
+        // let newHeaders = Object.assign(headers, {
+        //   Authorization: `Bearer ${(await Login.tokens()).access_token}`,
+        // });
+        // return newHeaders;
+
+        return headers;
+    }
+
+    static _buildQuery(params) {
+        if (params) {
+            return (
+                '?' +
+                Object.keys(params)
+                .map(k => encodeURIComponent(k) + '=' + encodeURIComponent(params[k]))
+                .join('&')
+            );
+        } else {
+            return '';
+        }
+    }
+
+    /**
+     * Perform a get method and add the appropriate headers.
+     * @param path : The path to get.  Required.
+     * @param query : the query parameters to append to the URL
+     * @param headers : extra headers to be included.  
+     * @returns a Promise the resolves the body of the response
+     */
+    static async get(path, query = {}, headers = {}) {
+        let fullHeaders = Object.assign(headers, await this._defaultHeaders());
+
+        let request = {
+            method: 'GET',
+            path: path + this._buildQuery(query),
+            headers: fullHeaders,
+        };
+        return this._call(request);
+    }
+
+    /**
+     * Perform a post method with the body and the appropriate headers.
+     * @param path : The path to post.  Required.
+     * @param body : the body of the request as an object
+     * @param headers : extra headers to be included.  
+     * @returns a Promise the resolves the body of the response
+     */
+    static async post(path, body = {}, headers = {}) {
+        let fullHeaders = Object.assign(headers, await this._defaultHeaders());
+        fullHeaders.Accept = 'application/json';
+        fullHeaders['Content-Type'] = 'application/x-www-form-urlencoded';
+
+        var formBody = [];
+        for (var property in body) {
+            var encodedKey = encodeURIComponent(property);
+            var encodedValue = encodeURIComponent(body[property]);
+            formBody.push(encodedKey + "=" + encodedValue);
+        }
+        formBody = formBody.join("&");
+
+        let request = {
+            method: 'POST',
+            path: path,
+            headers: fullHeaders,
+            body: formBody
+        };
+        return this._call(request);
+    }
+
+    /**
+     * Perform a patch method with the body and the appropriate headers.
+     * @param path : The path to get.  Required.
+     * @param body : the body of the request as an object
+     * @param headers : extra headers to be included.  
+     * @returns a Promise the resolves the body of the response
+     */
+    static async patch(path, id, body = {}, headers = {}) {
```

Comment:
If these methods are all static, does it make sense to just make this file a collection of functions? Is there a benefit to making them static methods?